File: easy/3541_find_most_frequent_vowel_and_consonant/solution.py
from collections import Counter


class Solution:
    def maxFreqSum(self, s: str) -> int:
        vowels_set = set("aeiou")

        max_vowel_count = 0
        max_consonant_count = 0

        for c, count in Counter(s).items():
            if c in vowels_set:
                if max_vowel_count < count:
                    max_vowel_count = count
            else:
                if max_consonant_count < count:
                    max_consonant_count = count

        return max_vowel_count + max_consonant_count

File: easy/3541_find_most_frequent_vowel_and_consonant/solution.cpp
using namespace std;

#include <string>
#include <unordered_map>

class Solution {
public:
  int maxFreqSum(string s) {
    unordered_map<char, int> chCounter;

    for (char &ch : s)
      chCounter[ch] += 1;

    int maxVowelCount = 0;
    int maxConsonantCount = 0;

    for (const auto &[ch, count] : chCounter) {
      if (ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u') {
        if (maxVowelCount < count)
          maxVowelCount = count;
      } else {
        if (maxConsonantCount < count)
          maxConsonantCount = count;
      }
    }

    return maxVowelCount + maxConsonantCount;
  }
};
